workos/resources/list.py

```diff
@@ -90,3 +107,66 @@ def auto_paging_iter(self):
                 next_page_marker = response["list_metadata"][string_direction]
                 yield data
                 data = []
+
+
+# add all possible generics of List Resource
+T = TypeVar("T", Organization, Directory, DirectoryGroup, DirectoryUser)
+
+
+class ListMetadata(BaseModel):
+    after: Optional[str] = None
+    before: Optional[str] = None
+
+
+class ListPage(BaseModel, Generic[T]):
+    object: Literal["list"]
+    data: List[T]
+    list_metadata: ListMetadata
+
+
+class ListArgs(BaseModel, extra="allow"):
+    limit: Optional[int] = 10
+    before: Optional[str] = None
+    after: Optional[str] = None
+    order: Literal["asc", "desc"] = "desc"
+
+    class Config:
+        extra = "allow"
+
+
+class WorkOsListResource(BaseModel, Generic[T]):
+    object: Literal["list"]
+    data: List[T]
+    list_metadata: ListMetadata
+
+    # These fields end up exposed in the types. Does we care?
+    list_method: Callable = Field(exclude=True)
+    list_args: ListArgs = Field(exclude=True)
```

Comment:
Where do these fields this show up in types exactly?

Reply:
They would appear a users IDE, like this
![CleanShot 2024-07-22 at 13 29 42](https://github.com/user-attachments/assets/011db569-87b5-4013-bb85-99759682dade)